doc/user/content/sql/recursive-ctes.md

```diff
@@ -0,0 +1,74 @@
+---
+title: Recursive CTEs
+description: "Recursive CTEs is used to define recursive queries."
+menu:
+  main:
+    parent: 'transform'
+    name: Recursive CTEs
+    weight: 22
+---
+
+
+Recursive CTEs can be used to define computations on recursively defined strucutes (such as trees or graphs) implied by the data in your tables.
+
+## Syntax
+
+### with_recursive_cte
+
+{{< diagram "with-recursive-ctes.svg" >}}
+
+### recursive_cte_binding
+
+{{< diagram "recursive-cte-binding.svg" >}}
+
+Field | Use
+------|-----
+**RETURN AT RECURSION LIMIT $n** | An optional clause indicating that the fixpoint computation should stop after `$n` iterations and offer the current values computed for each `recurive_cte_binding` to the `select_stmt`. This could be useful when debugging and validating the correctness of recursive queries. See the [Examples](#examples)
+**ERROR AT RECURSION LIMIT $n** |  An optional clause indicating that the fixpoint computation should stop after `$n` iterations fail the query with an error. Adding this clause with a reasonably high limit is a good safeguard against accidentally running a dataflow in your production clusters.
```

Comment:
- _and_ fail
- "running a dataflow" -> "running a _divergent_ dataflow" (or maybe say something like infinite loop, to avoid the maybe-scary word "divergent")

Reply:
Maybe "non-terminating" dataflow?